tests/secc/states/test_iso15118_2_states.py

```diff
@@ -77,3 +79,110 @@ async def test_authorization_next_state_on_authorization_request(
         authorization = Authorization(self.comm_session)
         authorization.process_message(message=get_dummy_v2g_message_authorization_req())
         assert authorization.next_state == expected_next_state
+
+    async def test_charge_parameter_discovery_res_v2g2_303(self):
+        # V2G2-303 : Sum of individual time intervals shall match the period of time
+        # indicated by the EVCC.
+        charge_parameter_discovery = ChargeParameterDiscovery(self.comm_session)
+
+        charge_parameter_discovery_req_departure_time_set = (
+            get_charge_parameter_discovery_req_message_departure_time_one_hour()
+        )
+        charge_parameter_discovery.process_message(
+            message=charge_parameter_discovery_req_departure_time_set
+        )
+
+        charging_duration = (
+            charge_parameter_discovery_req_departure_time_set.body.charge_parameter_discovery_req.ac_ev_charge_parameter.departure_time  # noqa
+        )
+
+        assert (
+            charge_parameter_discovery.next_msg.body.charge_parameter_discovery_res
+            is not None
+        )
+        charge_parameter_discovery_res = (
+            charge_parameter_discovery.next_msg.body.charge_parameter_discovery_res
+        )
+        assert charge_parameter_discovery_res.sa_schedule_list is not None
+        sa_schedule_tuples = (
+            charge_parameter_discovery_res.sa_schedule_list.schedule_tuples
+        )
+        for schedule_tuples in sa_schedule_tuples:
+            assert schedule_tuples.p_max_schedule is not None
+            schedule_duration = 0
+            if schedule_tuples.p_max_schedule.schedule_entries is not None:
+                start_time = schedule_tuples.p_max_schedule.schedule_entries[
+                    0
+                ].time_interval.start
+
+            for entry in schedule_tuples.p_max_schedule.schedule_entries:
+                schedule_duration += entry.time_interval.start - start_time
+                if entry.time_interval.duration is not None:
+                    schedule_duration += entry.time_interval.duration
+                start_time = entry.time_interval.start
```

Comment:
following the same comment as before, I think we can get the last and first entry and do the calculation

Reply:
Same as above - have done the mod.